alpaka CI: set CMAKE_CXX_COMPILER instead using update-alternatives

SimeonEhrig opened the pull request:
Changing the OS host compiler has dangerous side effects and is not common on productive systems. Therefore our CI should not do it. Instead the CI should use  `CMAKE_CXX_COMPILER` to set the compiler.

- Instead changing the default host compiler, set the compiler via CMAKE_CXX_COMPILER argument at CMake configure.
- Rename environment variable CXX to ALPAKA_CI_CXX, because CXX can cause unexpected side effects.
- The environment variable CMAKE_CXX_COMPILER is created in the scripts depending on the variable ALPAKA_CI_CXX.
- Remove the environment variable CC completely because we do not compile C code.
- Set also CMAKE_CUDA_COMPILER as CMake configure argument manually.

Set `CMAKE_CUDA_COMPILER` as CMake argument was also necessary because of the changed behavior of setting the C++ compiler.

psychocoderHPC commented on script/docker_ci.sh:
```diff
@@ -12,8 +12,8 @@ ALPAKA_CI_BOOST_BRANCH="boost-${ALPAKA_BOOST_VERSION}"
 ALPAKA_DOCKER_ENV_LIST=()
-ALPAKA_DOCKER_ENV_LIST+=("--env" "CC=${CC}")
-ALPAKA_DOCKER_ENV_LIST+=("--env" "CXX=${CXX}")
+ALPAKA_DOCKER_ENV_LIST+=("--env" "ALPAKA_CI_CC=${ALPAKA_CI_CC}")
```
setting the C compiler could be skipped we do not use the C compiler

SimeonEhrig replied:
The boost install script is the only one, which use the variable `ALPAKA_CI_CC`. I can check, if we can use the `ALPAKA_CI_CXX` instead.

SimeonEhrig replied:
I use the `ALPAKA_CI_CXX` variable to select the C compiler in the `install_boost.sh` script. Therefore I removed the variable `ALPAKA_CI_CC` and also `CC`.

psychocoderHPC commented on script/gitlabci/job_base.yml:
```diff
@@ -18,8 +18,8 @@
     ALPAKA_CI_UBUNTU_VER: "20.04"
-    CC: gcc
-    CXX: g++
+    ALPAKA_CI_CC: gcc
```
setting the C compiler should not be required an will automaticlally fall back to the system C compiler, alpaka is not compiling C code.

SimeonEhrig committed:
alpaka CI: set CMAKE_CXX_COMPILER instead using update-alternatives

- Instead changing the default host compiler, set the compiler via CMAKE_CXX_COMPILER argument at CMake configure.
- Rename environment variable CXX to ALPAKA_CI_CXX, because CXX can cause unexpected side effects.
- The environment variable CMAKE_CXX_COMPILER is created in the scripts depending on the variable ALPAKA_CI_CXX.
- Remove the environment variable CC completely because we do not compile C code.
- Set also CMAKE_CUDA_COMPILER as CMake configure argument manually.

SimeonEhrig commented:
@psychocoderHPC ready to merge
